icebox/icebox.py

```diff
@@ -1901,6 +1982,70 @@ def parse_db(text, device="1k"):
+
+        "PLLTYPE_1":            (14, 21, "PLLCONFIG_1"),
+        "PLLTYPE_2":            (14, 21, "PLLCONFIG_3"),
+        "PLLTYPE_0":            (12, 21, "PLLCONFIG_5"),
+        "FEEDBACK_PATH_0":      (14, 21, "PLLCONFIG_5"),
+        "FEEDBACK_PATH_1":      (11, 21, "PLLCONFIG_9"),
+        "FEEDBACK_PATH_2":      (12, 21, "PLLCONFIG_1"),
+        "PLLOUT_SELECT_A_0":    (12, 21, "PLLCONFIG_6"),
+        "PLLOUT_SELECT_A_1":    (12, 21, "PLLCONFIG_7"),
+        "PLLOUT_SELECT_B_0":    (12, 21, "PLLCONFIG_2"),
+        "PLLOUT_SELECT_B_1":    (12, 21, "PLLCONFIG_3"),
+        "SHIFTREG_DIV_MODE":    (12, 21, "PLLCONFIG_4"),
+        "FDA_FEEDBACK_0":       (12, 21, "PLLCONFIG_9"),
+        "FDA_FEEDBACK_1":       (13, 21, "PLLCONFIG_1"),
+        "FDA_FEEDBACK_2":       (13, 21, "PLLCONFIG_2"),
+        "FDA_FEEDBACK_3":       (13, 21, "PLLCONFIG_3"),
+        "FDA_RELATIVE_0":       (13, 21, "PLLCONFIG_5"),
+        "FDA_RELATIVE_1":       (13, 21, "PLLCONFIG_6"),
+        "FDA_RELATIVE_2":       (13, 21, "PLLCONFIG_7"),
+        "FDA_RELATIVE_3":       (13, 21, "PLLCONFIG_8"),
+        "DIVR_0":               (10, 21, "PLLCONFIG_1"),
+        "DIVR_1":               (10, 21, "PLLCONFIG_2"),
+        "DIVR_2":               (10, 21, "PLLCONFIG_3"),
+        "DIVR_3":               (10, 21, "PLLCONFIG_4"),
+        "DIVF_0":               (10, 21, "PLLCONFIG_5"),
+        "DIVF_1":               (10, 21, "PLLCONFIG_6"),
+        "DIVF_2":               (10, 21, "PLLCONFIG_7"),
+        "DIVF_3":               (10, 21, "PLLCONFIG_8"),
+        "DIVF_4":               (10, 21, "PLLCONFIG_9"),
+        "DIVF_5":               (11, 21, "PLLCONFIG_1"),
+        "DIVF_6":               (11, 21, "PLLCONFIG_2"),
+        "DIVQ_0":               (11, 21, "PLLCONFIG_3"),
+        "DIVQ_1":               (11, 21, "PLLCONFIG_4"),
+        "DIVQ_2":               (11, 21, "PLLCONFIG_5"),
+        "FILTER_RANGE_0":       (11, 21, "PLLCONFIG_6"),
+        "FILTER_RANGE_1":       (11, 21, "PLLCONFIG_7"),
+        "FILTER_RANGE_2":       (11, 21, "PLLCONFIG_8"),
+        "TEST_MODE":            (12, 21, "PLLCONFIG_8"),
+        "DELAY_ADJMODE_FB":     (13, 21, "PLLCONFIG_4"),
+        "DELAY_ADJMODE_REL":    (13, 21, "PLLCONFIG_9"),
+
+        # PLL Ports
+        "PLLOUT_A":             ( 12, 21, 1),
+        "PLLOUT_B":             ( 13, 21, 0),
+        "REFERENCECLK":         ( 10, 21, "fabout"),
+        "EXTFEEDBACK":          ( 11, 21, "fabout"),
+        "DYNAMICDELAY_0":       (  1, 21, "fabout"),
+        "DYNAMICDELAY_1":       (  2, 21, "fabout"),
+        "DYNAMICDELAY_2":       (  3, 21, "fabout"),
+        "DYNAMICDELAY_3":       (  4, 21, "fabout"),
+        "DYNAMICDELAY_4":       (  5, 21, "fabout"),
+        "DYNAMICDELAY_5":       (  7, 21, "fabout"),
+        "DYNAMICDELAY_6":       (  8, 21, "fabout"),
+        "DYNAMICDELAY_7":       (  9, 21, "fabout"),
+        "LOCK":                 (  1, 20, "neigh_op_tnl_1"), #check?
```

Comment:
These look correct to me - if you are curious, the PLL status outputs come from otherwise non-existent neighbour connections - in this case above and left of tile (1, 20) - the corner of the device where there are no LUTs.

Reply:
ah!  that makes sense